tests/integration/test_client_server.py

```diff
@@ -109,3 +110,24 @@ def test_endpoint_not_found(mock_server):
             endpoint='fooo',
             hostport=mock_server.hostport,
         )
+
+
+@pytest.mark.gen_test
+def test_connection_close(mock_server):
+    tchannel = TChannel(name='test')
+
+    class TestHook(EventHook):
+        def before_send_request(self, request):
+            # close the connection
+            peer = tchannel._dep_tchannel.peers.get(mock_server.hostport)
+            peer.close()
+            peer.connections[0].closed = True
+
+    tchannel.hooks.register(TestHook())
```

Comment:
Also, just to confirm: Is this test just trying to verify that we raise a `NetworkError ` when the destination doesn't exist? Nothing to do with callbacks, right?